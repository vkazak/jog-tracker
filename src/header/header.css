.header {
    background-color: #7ed321;
    height: 77px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
    padding: 0px 25px;
}

.header-on-menu-opened {
    background-color: #ffffff;
}

.logo {
    content: url('logo.png');
    height: 37px;
    margin-right: auto;
    width: auto;
}

.logo-on-menu-opened {
    content: url('logo-green.png');
}

.nav-container {
    position: absolute; 
    top: 77px;
    left: 0px;
    display: none;
    width: 100vw;
    height: calc(100vh - 77px);
    background-color: white;
    z-index: 2;
}

.nav-container-on-menu-opened {
    display: flex;
}

.nav-items-box {
    margin: 30% auto;
    display: flex;
    height: 170px;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

.nav-item {
    display: block;
    color: #000000;
    text-decoration: none;
    font-weight: bold;
    font-size: 25px;
    text-transform: uppercase;
}

.menu-button {
    border: none;
    background-color: transparent;
    color: white;
    cursor: pointer;
    font-size: 28px;
}

.filter-button {
    border: none;
    background-color: transparent;
    color: white;
    cursor: pointer;
    font-size: 28px;
    margin-right: 39px;
}

.filter-button-active {
    color: #62aa14;
    border-style: solid;
    border-color: #62aa14;
    border-radius: 5px;
}

.menu-button-on-menu-opened {
    color: #b0abab;
}

.hidden {
    display: none;
}

@media screen and (min-width: 600px) {
    .header {
        height: 116px;
    }
    .logo {
        height: 56px;
    }
    .menu-button {
        display: none;
    }
    .filter-button {
        margin-right: 0px;
        margin-left: 56px;
    }
    .nav-container {
        position: static;
        display: block;
        background-color: transparent;
        width: 100%;
        height: 100%;
    }
    .nav-items-box {
        flex-direction: row;
        justify-content: flex-end;
        height: 100%;
        margin: 0px;
    }
    .nav-item {
        color: #ffffff;
        margin-left: 45px;
        font-size: 14px;
    }
}