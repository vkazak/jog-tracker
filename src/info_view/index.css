.info-container {
    margin: 25px;
    max-width: 614px;
}

h1 {
    color: #7ed321;
    font-size: 25px;
    text-transform: uppercase;
    margin-bottom: 6px;
}

.info-paragraph {
    font-size: 14px;
    line-height: 24px;
    margin-bottom: 30px;
}

@media screen and (min-width: 600px) {
    .info-container {
        margin: auto;
        margin-top: 85px;
    }
    h1 {
        font-size: 36px;
        margin-bottom: 15px;
    }
}