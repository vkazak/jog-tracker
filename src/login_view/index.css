.container {
    display: flex;
    height: calc(100vh - 77px);
    justify-content: center;
    align-items: center;
    overflow: auto;
}

.outer-rectangle {
    border-radius: 44px;
}

.content-col {
    width: 160px;
    margin: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.bear-image {
    width: 100%;
    content: url('bear-face.png');
    height: auto;
    margin-bottom: 103px;
}

.let-me-button {
    height: 60px;
    border-radius: 30px;
    border-width: 3px;
    border-color: #e990f9;
    border-style: solid;
    color: #e990f9;
    font-weight: bold;
    font-size: 18px;
    background-color: #ffffff;
    margin: 0px 5px;
}

@media screen and (min-width: 600px) {
    .container {
        height: calc(100vh - 116px);
    }
    .outer-rectangle {
        height: 379px;
        width: 503px;
        background-color: #e990f9;
    }
    .bear-image {
        content: url('bear-face-white.svg');
        margin: 77px auto 56px;
    }
    .let-me-button {
        border-color: #ffffff;
        background-color: #e990f9;
        color: #ffffff;
    }
}