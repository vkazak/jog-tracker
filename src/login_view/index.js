import React from 'react';
import './index.css';
import { loginByUUID } from '../request/request';
import { useHistory } from 'react-router-dom';

export function LogInView(props) {
    const history = useHistory();
    
    const letMeIn = async () => {
        try {
            const uuid = 'hello';
            const token = await loginByUUID(uuid);
            localStorage.setItem('token', token);
            console.log(token);
            history.push('/jogs');
        } catch(err) {
            console.log(err);
        }
    }

    return (
        <div className='container'>
            <div className='outer-rectangle'>
                <div className='content-col'>
                    <img className='bear-image' alt='Bear face' />
                    <button className='let-me-button' onClick={letMeIn}> 
                        Let me in 
                    </button> 
                </div>
            </div>
        </div>
    )
}