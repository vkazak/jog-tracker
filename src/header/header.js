import React, { useState } from 'react';
import './header.css';

function MenuItem(props) {
    
    return(
        <a className="nav-item" href={props.link}>
            {props.title} 
        </a>
    )
}

export function Header(props) {
    const [menuOpened, setMenuOpened] = useState(false);

    const menuItems = [
        { title: 'jogs', link: '/jogs' },
        { title: 'info', link: '/info' },
        { title: 'contact us', link: '/contact_us' }
    ];

    let headerClass = 'header'; 
    let navContainerClass = 'nav-container';
    let logoClass = 'logo';
    let menuButtonClass = 'menu-button';
    let buttonIconClass = 'fa fa-bars';
    if (menuOpened) {
        headerClass += ' header-on-menu-opened';
        navContainerClass += ' nav-container-on-menu-opened';
        logoClass += ' logo-on-menu-opened';
        menuButtonClass += ' menu-button-on-menu-opened';
        buttonIconClass = 'fa fa-close';
    }

    return (
        <div className={headerClass}>
            <img className={logoClass} alt='Logo' />
            <div className={navContainerClass}> 
                <div className='nav-items-box'>
                    { menuItems.map(item => <MenuItem {...item}/>) }
                </div>
            </div>
            <button className={menuButtonClass} onClick={ () => setMenuOpened(!menuOpened) }>
                <i className={buttonIconClass}></i>
            </button>
        </div>
    )
}