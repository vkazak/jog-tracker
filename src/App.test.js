import React from 'react';
import renderer from 'react-test-renderer';
import { MenuItem } from './header/header';
import { InfoView } from './info_view';
import { LogInView } from './login_view';

it('renders header menu item correctly', () => {
  const tree = renderer.create(<MenuItem title='test' link='/test'/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders info view correctly', () => {
  const tree = renderer.create(<InfoView />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders login view correctly', () => {
  const tree = renderer.create(<LogInView />).toJSON();
  expect(tree).toMatchSnapshot();
})