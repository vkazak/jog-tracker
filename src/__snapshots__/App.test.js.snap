// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header menu item correctly 1`] = `
<a
  className="nav-item"
  href="/test"
  onClick={[Function]}
>
  test
</a>
`;

exports[`renders info view correctly 1`] = `
<div
  className="info-container"
>
  <h1>
    Info
  </h1>
  <p
    className="info-paragraph"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing  software like Aldus PageMaker including versions of Lorem Ipsum.
  </p>
  <p
    className="info-paragraph"
  >
    It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English.
  </p>
</div>
`;

exports[`renders login view correctly 1`] = `
<div
  className="container"
>
  <div
    className="outer-rectangle"
  >
    <div
      className="content-col"
    >
      <img
        alt="Bear face"
        className="bear-image"
      />
      <button
        className="let-me-button"
        onClick={[Function]}
      >
        Let me in
      </button>
    </div>
  </div>
</div>
`;
